### PR TITLE
Update M3 library, Increase target sdk to 34

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -84,8 +84,8 @@ android {
 
     defaultConfig {
         minSdkVersion 24
-        targetSdkVersion 33
-        compileSdk 33
+        targetSdkVersion 34
```
Comment (alperozturk96) [inline app/build.gradle:87]: Needed for latest M3 library

**Review by ZetaTom — COMMENTED** [review]
Seems to work well during my brief tests. Have the [behavioural changes](https://developer.android.com/about/versions/14/behavior-changes-all) of Android 14 been addressed already?

According to the lint bot there are apparently some missing `super` calls in various activities that haven't been changed by this pull request. Maybe something else or some requirements have changed in the meantime?

**Comment by alperozturk96** [comment]
As far as I know regarding behavioural changes on Android 14 nothing specificly done in codebase. Regardless targetSdkVersion if device have Android 14 that changes automatically done so increasing or decreasing targetSdkVersion not going to effect anything on Android 14 devices. @ZetaTom We must check "Behavior changes" for Android 14 device but I think it's not related to this PR. 

Seems like after targetSdkVersion change Linter seeing those intentionally not called super method as warning. However, I see that the default action is not requested in those activities. If we call super methods, the behavior of the application may change. In this case we can use @Suppress annotation.

**Comment by tobiasKaminsky** [comment]
Unit tests are failing to compile:
> Task :app:compileGplayDebugUnitTestKotlin FAILED
e: file:///home/runner/actions-runner/_work/android/android/app/src/test/java/com/***/client/core/LocalConnectionTest.kt:68:39 Overload resolution ambiguity: 
public open fun bindService(p0: Intent, p1: ServiceConnection, p2: Context.BindServiceFlags): Boolean defined in android.content.Context
public abstract fun bindService(p0: Intent, p1: ServiceConnection, p2: Int): Boolean defined in android.content.Context
e: file:///home/runner/actions-runner/_work/android/android/app/src/test/java/com/***/client/core/LocalConnectionTest.kt:79:25 None of the following functions can be called with the arguments supplied: 
public open fun bindService(p0: Intent, p1: ServiceConnection, p2: Context.BindServiceFlags): Boolean defined in android.content.Context
public abstract fun bindService(p0: Intent, p1: ServiceConnection, p2: Int): Boolean defined in android.content.Context
e: file:///home/runner/actions-runner/_work/android/android/app/src/test/java/com/***/client/core/LocalConnectionTest.kt:79:73 The boolean literal does not conform to the expected type Unit
e: file:///home/runner/actions-runner/_work/android/android/app/src/test/java/com/***/client/core/LocalConnectionTest.kt:84:26 None of the following functions can be called with the arguments supplied: 
public open fun bindService(p0: Intent, p1: ServiceConnection, p2: Context.BindServiceFlags): Boolean defined in android.content.Context
public abstract fun bindService(p0: Intent, p1: ServiceConnection, p2: Int): Boolean defined in android.content.Context
w: Detected multiple Kotlin daemon sessions at kotlin/sessions

**Comment by tobiasKaminsky** [comment]
Android 14 specific can and should be discussed in https://github.com/nextcloud/android/issues/11791

**Review by tobiasKaminsky — CHANGES_REQUESTED** [review]
Tests are failing to compile

**Comment by AndyScherzinger** [comment]
@alperozturk96 has conflicts on rebase (!)

**Comment by AndyScherzinger** [comment]
@alperozturk96 you need to rebase yourself, since you will have to resolve conflicts which you originally resolved during pulling in master. So the rebase comment won't work.

**Review by AndyScherzinger — APPROVED** [review]
Unit tests are fine, only codecov upload fails (not related) and Drone failures are also not related.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @ZetaTom works for my on my phone, however I didn't check for the behavioral changed but as stated by Tobias are to be discussed in the or ticket. So for me the PR is good to go.

**Pull request merged**